pgrx-tests/src/tests/geo_tests.rs

```diff
@@ -13,24 +13,97 @@ mod tests {
     #[allow(unused_imports)]
     use crate as pgrx_tests;
 
+    use pgrx::datum::geo::*;
     use pgrx::prelude::*;
 
     #[pg_test]
-    fn test_point_into_datum() -> spi::Result<()> {
-        let p =
-            Spi::get_one::<pg_sys::Point>("SELECT '42, 99'::point")?.expect("SPI result was null");
-        assert_eq!(p.x, 42.0);
-        assert_eq!(p.y, 99.0);
+    fn test_point_datum() -> spi::Result<()> {
+        let p = Spi::get_one::<Point>("SELECT '42, 99'::point")?.expect("SPI result was null");
+        assert_eq!(p, Point { x: 42.0, y: 99.0 });
+        let p2 = Spi::get_one_with_args::<Point>(
+            "SELECT $1",
+            vec![(Point::type_oid().into(), p.into_datum())],
+        )?
+        .expect("SPI result was null");
+        assert_eq!(p, p2);
         Ok(())
     }
 
     #[pg_test]
-    fn test_box_into_datum() -> spi::Result<()> {
-        let b = Spi::get_one::<pg_sys::BOX>("SELECT '1,2,3,4'::box")?.expect("SPI result was null");
-        assert_eq!(b.high.x, 3.0);
-        assert_eq!(b.high.y, 4.0);
-        assert_eq!(b.low.x, 1.0);
-        assert_eq!(b.low.y, 2.0);
+    fn test_box_datum() -> spi::Result<()> {
+        let b = Spi::get_one::<Box>("SELECT '1,2,3,4'::box")?.expect("SPI result was null");
+        assert_eq!(b, Box { high: Point { x: 3.0, y: 4.0 }, low: Point { x: 1.0, y: 2.0 } });
+        let b2 = Spi::get_one_with_args::<Box>(
+            "SELECT $1",
+            vec![(Box::type_oid().into(), b.into_datum())],
+        )?
+        .expect("SPI result was null");
+        assert_eq!(b, b2);
+        Ok(())
+    }
+
+    #[pg_test]
+    fn test_lseg_datum() -> spi::Result<()> {
```

Comment:
These tests look good, but can we also get one or two that show the same types being passed around using the `fn_call` module? I've noticed sometimes there's odd discrepancies between various ways of moving data in/out using literals and SPI, and in/out using other interfaces. You don't have to do all of them, just whichever type seems most quirky.